packages/core/test/scxml.test.ts

```diff
@@ -137,194 +138,205 @@ const testGroups = {
+    // 'test239.txml', // done.invoke used as inexact event descriptor
+    // 'test240.txml', // conversion of namelist not implemented yet
+    // 'test241.txml', // conversion of namelist not implemented yet
+    // 'test242.txml', // <invoke src="...">
+    // 'test243.txml', // conversion of <param> in <scxml> not implemented yet
+    // 'test244.txml', // conversion of namelist not implemented yet
+    // 'test245.txml', // conversion of namelist not implemented yet
+    // 'test247.txml', // done.invoke used as inexact event descriptor
+    // 'test250.txml', // this is a manual test - we could test it by snapshoting logged valued
+    // 'test252.txml', // done.invoke used as inexact event descriptor
+    // 'test253.txml', // _event.origintype not implemented yet
+    // 'test276.txml', // <invoke src="...">
+    // 'test277.txml', // illegal expression in datamodel creates unbound variable
+    // 'test278.txml', // non-root datamodel with early binding not implemented yet
+    // 'test279.txml', // non-root datamodel with early binding not implemented yet
+    // 'test280.txml', // non-root datamodel with late binding not implemented yet
+    // 'test286.txml', // error.execution when evaluating assign
     'test287.txml',
-    // 'test294.txml',
-    // 'test298.txml',
-    // 'test302.txml',
-    // 'test303-1.txml',
-    // 'test303-2.txml',
-    // 'test303.txml',
-    // 'test304.txml',
-    // 'test307.txml',
-    // 'test309.txml',
-    // 'test310.txml',
-    // 'test311.txml',
-    // 'test312.txml',
-    // 'test313.txml',
-    // 'test314.txml',
+    // 'test294.txml', // conversion of <donedata> not implemented yet
+    // 'test298.txml', // error.execution when evaluating donedata
+    // 'test302.txml', // conversion of <script> not implemented yet
+    // 'test303-1.txml', // conversion of <script> not implemented yet
+    // 'test303-2.txml', // conversion of <script> not implemented yet
+    // 'test303.txml', // conversion of <script> not implemented yet
+    // 'test304.txml', // conversion of <script> not implemented yet
+    // 'test307.txml', // non-root datamodel with late binding not implemented yet
+    // 'test309.txml', // error in cond expression being treated as false
+    // 'test310.txml', // conversion of In() predicate not implemented yet
+    // 'test311.txml', // error.execution when evaluating assign
+    // 'test312.txml', // error.execution when evaluating assign
+    // 'test313.txml', // error.execution when evaluating assign
+    // 'test314.txml', // error.execution when evaluating assign
     'test318.txml',
-    // 'test319.txml',
-    // 'test321.txml',
-    // 'test322.txml',
-    // 'test323.txml',
-    // 'test324.txml',
-    // 'test325.txml',
-    // 'test326.txml',
-    // 'test329.txml',
-    // 'test330.txml',
-    'test331.txml',
-    // 'test332.txml',
+    // 'test319.txml', // SCXML has no init event, so _event stays unbound in onentry of initial state
```

Comment:
Makes me wonder if we should get rid of the `xstate.init` event (as you suggested).

I would be fine having a `null as any` as the event, and allowing it to throw if the developer happens to try to use the `event` in any entry action on the initial state (which they shouldn't do).

Reply:
Definitely something to discuss. One problem with removing it is that we:
- either would have to annotate `event` as optional inside actions (especially those defined on the `options.actions`
- or annotate them as TEvent and let it crash at runtime if one access it

Introducing "XState type compiler" would just fix this issue (given people would set it up), but I doubt we are able to ship something like that together with v5 release - it's rather a bigger project & enhancement that we could explore further down the road.

I'm not opposed to removing the init event, just mentioning the downsides of doing so.

Reply:
I would prefer annotating it and letting it crash. We can even catch it and provide a descriptive error message, as the developer should not try to access the event initially anyway.